### ISSUE TITLE
Pack should also pack project references (recursive pack)

### ISSUE THREAD

**Issue by zivkan** [NuGet/Home#13436]
### NuGet Product(s) Involved

dotnet.exe

### The Elevator Pitch

If you do a `dotnet restore my.csproj` or `dotnet build my.csproj`, it will recursively restore or build all the projects.   `dotnet pack` should do the same.

```
dotnet new classlib -n ClassLib1
dotnet new classlib -n ClassLib2
dotnet add ClassLib1/ClassLib1.csproj reference ClassLib2
dotnet pack ClassLib1
```

The above should create nupkgs for both projects, not just ClassLib1. We can see that ClassLib2 was restored and built, just not packed.

### Additional Context and Details

While `dotnet publish` is similar to `dotnet pack`'s current behaviour, it does not recursively publish all referenced projects, I believe that's fine because the output of referenced projects are included in the target project's output. Therefore, I don't think that `dotnet publish` is "prior art" in having non-recursive behaviour as pack is not similar enough to publish.

**Comment by AArnott** [comment]
Yes, this has bitten me for years. the ClassLib1 package isn't valid by itself, since it references a non-existent package.

**Comment by Aaronontheweb** [comment]
Doesn't `dotnet pack` on the `.sln` already do most of this?

**Comment by AArnott** [comment]
`dotnet pack` on the sln just invokes `pack` on each project in the solution. It isn't recursive, but if your solution is fully-inclusive of your projects, then it packs everything alright.
This bug is about `dotnet pack some.csproj` though, since that leaves an incomplete build of packages that you may not be able to use. Not all repos have sln files. And those that do may still be so big you don't want to pack the world when you only want to pack a few projects.

**Comment by AArnott** [comment]
I'd still like to see this.
